moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -859,6 +870,46 @@ bool TrajectoryExecutionManager::distributeTrajectory(const moveit_msgs::RobotTr
   return true;
 }
 
+bool TrajectoryExecutionManager::validate(const moveit_msgs::RobotTrajectory &trajectory) const
+{
+  ROS_DEBUG_NAMED("traj_execution", "validating trajectory with allowed_start_tolerance %g", allowed_start_tolerance_);
+
+  if (trajectory.joint_trajectory.points.empty())
+    return true;
+
+  robot_state::RobotStatePtr current_state;
+  if (!csm_->waitForCurrentState(1.0) || !(current_state = csm_->getCurrentState()))
+  {
+    ROS_WARN_NAMED("traj_execution", "Failed to validate trajectory: couldn't receive full current joint state within 1s");
+    return false;
+  }
+
+  const std::vector<double> &positions = trajectory.joint_trajectory.points.front().positions;
+  const std::vector<std::string> &joint_names = trajectory.joint_trajectory.joint_names;
+  const std::size_t n = joint_names.size();
+  if (positions.size() != n)
+    return false;
+
+  for (std::size_t i = 0; i < n; ++i)
+  {
+    const robot_model::JointModel *jm = current_state->getJointModel(joint_names[i]);
+    if (!jm)
+    {
+      ROS_ERROR_STREAM_NAMED("traj_execution", "Unknown joint in trajectory: " << joint_names[i]);
+      return false;
+    }
+    // TODO: check multi-DoF joints ?
+    if (fabs(current_state->getJointPositions(jm)[0] - positions[i]) > allowed_start_tolerance_)
+    {
+      ROS_ERROR_NAMED("traj_execution",
+                      "Invalid Trajectory: start point deviates from current robot state more than %g",
```

Comment:
Please add the joint name and the expected / current values here to support debugging.